src/arch/MovieTexture/MovieTexture_Generic.cpp

```diff
@@ -303,189 +315,90 @@ void MovieTexture_Generic::CreateTexture()
  */
 float MovieTexture_Generic::CheckFrameTime()
 {
-	if( m_fRate == 0 )
+	if (m_fRate == 0)
 		return 1;	// "a long time until the next frame"
 
-	const float fOffset = (m_pDecoder->GetTimestamp() - m_fClock) / m_fRate;
-
-	/* If we're ahead, we're decoding too fast; delay. */
-	if( fOffset > 0.00001f )
-	{
-		if( m_bFrameSkipMode )
-		{
-			/* We're caught up; stop skipping frames. */
-			LOG->Trace( "stopped skipping frames" );
-			m_bFrameSkipMode = false;
-		}
-		return fOffset;
-	}
-
-	/*
-	 * We're behind by -Offset seconds.
-	 *
-	 * If we're just slightly behind, don't worry about it; we'll simply
-	 * not sleep, so we'll move as fast as we can to catch up.
-	 *
-	 * If we're far behind, we're short on CPU.  Skip texture updates; this
-	 * is a big bottleneck on many systems.
-	 *
-	 * If we hit a threshold, start skipping frames via #1.  If we do that,
-	 * don't stop once we hit the threshold; keep doing it until we're fully
-	 * caught up.
-	 *
-	 * We should try to notice if we simply don't have enough CPU for the video;
-	 * it's better to just stay in frame skip mode than to enter and exit it
-	 * constantly, but we don't want to do that due to a single timing glitch.
-	 */
-	const float FrameSkipThreshold = 0.5f;
-
-	if( -fOffset >= FrameSkipThreshold && !m_bFrameSkipMode )
-	{
-		LOG->Trace( "(%s) Time is %f, and the movie is at %f.  Entering frame skip mode.",
-			GetID().filename.c_str(), m_fClock, m_pDecoder->GetTimestamp() );
-		m_bFrameSkipMode = true;
-	}
 
-	return 0;
+	return (m_pDecoder->GetTimestamp() - m_fClock) / m_fRate;
 }
 
-/* Decode data. */
-void MovieTexture_Generic::DecodeSeconds( float fSeconds )
+void MovieTexture_Generic::UpdateMovie(float fSeconds)
 {
+	if (m_failure) return;
 	m_fClock += fSeconds * m_fRate;
 
-	/* We might need to decode more than one frame per update.  However, there
-	 * have been bugs in ffmpeg that cause it to not handle EOF properly, which
-	 * could make this never return, so let's play it safe. */
-	int iMax = 4;
-	while( --iMax )
-	{
-		/* If we don't have a frame decoded, decode one. */
-		if( m_ImageWaiting == FRAME_NONE )
-		{
-			if( !DecodeFrame() )
-				break;
-
-			m_ImageWaiting = FRAME_DECODED;
-		}
-
-		/* If we have a frame decoded, see if it's time to display it. */
-		float fTime = CheckFrameTime();
-		if ( fTime <= 0 )
-		{
-			UpdateFrame();
-			m_ImageWaiting = FRAME_NONE;
-		}
+	// If the frame isn't ready, don't update. This does mean the video
+	// will "speed up" to catch up when decoding does outpace display.
+	//
+	// In practice, display should rarely, if ever, outpace decoding.
+	if (m_pDecoder->IsCurrentFrameReady() && CheckFrameTime() <= 0) {
+		UpdateFrame();
 		return;
 	}
-
-	LOG->MapLog( "movie_looping", "MovieTexture_Generic::Update looping" );
 }
 
 void MovieTexture_Generic::UpdateFrame()
 {
 	/* Just in case we were invalidated: */
 	CreateTexture();
 
-	if( m_pTextureLock != nullptr )
+	if (m_pTextureLock != nullptr)
```

Comment:
whitespace inconsistency

Reply:
Just to be clear, what's the expected style here? I was trying to move things more toward what I expect when looking at C++ style, which is to say the spaces on either side of the parentheses look out of place to me. I don't mind changing to match the general style of the repo--even if it looks a little off to me 🙂 

Or are you referring to spaces vs. tabs? Just want to fully understand what consistency you're looking for here. This question applies here and all the other places you marked inconsistent (spacing on ternary operators, spacing on parentheses, a space after variable assignment, etc...)

Reply:
All I meant is that it makes the diff a little harder to view when there are lots of little whitespace changes. ☺️

Reply:
Ah gotcha. I planned on breaking this PR into a bunch of smaller ones to hopefully make it easier to digest with the whitespace changes. Though it's fair for the whole of this PR to maybe exclude some of those whitespace changes if someone does want to look at the full context until it's all said and done. I'll do another lap on this eventually.